pkg/server/api/status.go

```diff
@@ -16,17 +18,60 @@ func CreateStatus(code codes.Code, format string, a ...interface{}) *types.Statu
 	}
 }
 
-// StatusFromError creates a proto Status from given error
-func StatusFromError(err error) *types.Status {
-	if err == nil {
+// OK creates a success proto status
+func OK() *types.Status {
+	return CreateStatus(codes.OK, codes.OK.String())
+}
+
+// MakeStatus logs and returns a status composed of: msg, err and code.
+// Errors are treated differentially according to its gRPC code.
+func MakeStatus(log logrus.FieldLogger, code codes.Code, msg string, err error) *types.Status {
+	e := MakeErr(log, code, msg, err)
+	if e == nil {
+		return OK()
+	}
+
+	return CreateStatus(code, status.Convert(e).Message())
+}
+
+// MakeErr logs and returns an error comoposed of: msg, err and code.
+// Errors are treated differentially according to its gRPC code.
+func MakeErr(log logrus.FieldLogger, code codes.Code, msg string, err error) error {
+	errMsg := msg
```

Comment:
errMsg is not really used, it is replaced after initialized

Reply:
`errMsg` is used in the `default` and `InvalidArgument` cases. In those cases the returned error contains the description of the inner error (if not nil).